lua/compe_ultisnips/init.lua

```diff
@@ -0,0 +1,53 @@
+local compe = require'compe'
+
+local M = {}
+
+function M:get_metadata()
+  return {
+    priority = 50,
+    dup = 0,
+    menu = '[Ultisnips]',
+  }
+end
+
+function M:datermine(context)
+  return compe.helper.datermine(context)
+end
+
+function M:complete(args)
+  if vim.fn.exists("*UltiSnips#SnippetsInCurrentScope") == 0 then
+    args.abort()
+    return
+  end
+  local receivedSnippets = vim.call('UltiSnips#SnippetsInCurrentScope')
+  if vim.tbl_isempty(receivedSnippets) then
+    args.abort()
+    return
+  end
+  local completionList = {}
+  for key, value in pairs(receivedSnippets) do
+    local item = {
+      word =  key,
+      abbr =  key,
+      userdata = value,
```

Comment:
it should be `user_data` because vim's completed-item uses it.

Reply:
OK! I didn't know about it. Just changing it. :smiley: